src/cs50/sql.py

```diff
@@ -316,8 +326,20 @@ def shutdown_session(exception=None):
                 # Join tokens into statement, abbreviating binary data as <class 'bytes'>
                 _statement = "".join([str(bytes) if token.ttype == sqlparse.tokens.Other else str(token) for token in tokens])
 
+                # Check for start of transaction
+                if command in ["BEGIN", "START"]:
+                    self._autocommit = False
+
                 # Execute statement
+                if self._autocommit:
+                    connection.execute(sqlalchemy.text("BEGIN"))
                 result = connection.execute(sqlalchemy.text(statement))
+                if self._autocommit:
+                    connection.execute(sqlalchemy.text("COMMIT"))
```

Comment:
Would that execute `BEGIN/START TRANSACTION` and `COMMIT/ROLLBACK` twice, once in the `if` statements and once on line 336?

Reply:
Shouldn't, because 

```py
                if command in ["BEGIN", "START"]:
                    self._autocommit = False
```

sets `self._autocommit` to `False`, which means the `if` won't evaluate to true in such cases?